### DISCUSSION TITLE
app_store_connect_api_key() -> error: invalid curve name

### DISCUSSION THREAD

**Discussion by RobTranquillo** [fastlane/fastlane#19768] [Q&A]
Hi,  
I can't get my Unity project to be built using game-ci's github/action. So I am trying to debug it on my local Mac using Fastlane.

But with `key_filepath` or `key_content` in base64 ruby crashes:
![grafik](https://user-images.githubusercontent.com/311739/148093280-941967d3-cf88-4d76-ab66-1f93056b2cf5.png)
![grafik](https://user-images.githubusercontent.com/311739/148093330-75e204aa-887f-4b9f-8fa1-c0b53d4adc22.png)

and with `key_content` it states an invald curve name:

```
vrbits@VRs-Mac-mini xxxx % bundle exec fastlane ios build
[✔] 🚀 
[17:28:08]: Driving the lane 'ios build' 🚀
[17:28:08]: ----------------------
[17:28:08]: --- Step: setup_ci ---
[17:28:08]: ----------------------
[17:28:08]: Not running on CI, skipping CI setup
[17:28:08]: ---------------------------------------
[17:28:08]: --- Step: app_store_connect_api_key ---
[17:28:08]: ---------------------------------------
+---------------+-----------+
|       Lane Context        |
+---------------+-----------+
| PLATFORM_NAME | ios       |
| LANE_NAME     | ios build |
+---------------+-----------+
[17:28:08]: invalid curve name

+------+---------------------------+-------------+
|                fastlane summary                |
+------+---------------------------+-------------+
| Step | Action                    | Time (in s) |
+------+---------------------------+-------------+
| 1    | setup_ci                  | 0           |
| 💥   | app_store_connect_api_key | 0           |
+------+---------------------------+-------------+
```

My Fastfile on this pos:
```
desc "Create .ipa"
lane :build do
    setup_ci
    app_store_connect_api_key(
        key_id: 'P8xxxxxxxx',
        issuer_id: '11111111-2222-3333-4444-555555555555',
        key_content: '-----BEGIN PRIVATE KEY-----\nMIGTAgEAMBxxxx\n-----END PRIVATE KEY-----'
    )
```

The example for http://docs.fastlane.tools/actions/app_store_connect_api_key/ shows that you use an elliptic curve: `BEGIN EC PRIVATE KEY`. But my key from Apple is only a RSA key: `-BEGIN PRIVATE KEY-`

**Comment by abdulbasit18** [comment]
Hey 

You should put base64 key instead of this hopefully it should work 
you can create the base64 using this command `cat key.p8 | base64`

FastFile should look like this 

```
    app_store_connect_api_key(
      key_id: *******],
      issuer_id: **********,
      key_content: ****************,
      is_key_content_base64: true,
      in_house: false #boolean value if team is Enterprise or not
    )

```

**Reply by RobTranquillo** [reply]
Hi and thank you for your answer. But my second image shows exactly that I was trying with base64 and it crashes. Or did i miss somthing?

**Reply by abdulbasit18** [reply]
@RobTranquillo Sorry I missed it, what I can i see is you have to add `in_house` property
